### DISCUSSION TITLE
Possible use nested functions in where query?

### DISCUSSION THREAD

**Discussion by martin5233** [skeeto/emacsql#118] [Support] [answered]
I was trying to use Emacsql for a query with a nested function call. The SQL statement would look something like:
`SELECT revision FROM revisions WHERE lower(hex(commit)) = "some hash value"`
where _revision_ and _commit_ are both columns in the table _revisions_.

I tried the following, which seems equivalent to me:
 ```
 (emacsql db [:select [ revision ]
                       :from revisions
                       :where (= (funcall lower (funcall hex commit)) "some hash value")]))
```

However this always returns nil, although I'm sure that the equivalent SQL statement returns a value. What am I doing wrong? Or isn't it possible to use nested functions for the SQL statement?

**Comment by tarsius** [comment, marked as answer]
Place point after the last `]` and `M-x emacsql-show-last-sql`. The result shows in `*emacsql-show*` or `*Messages*`:

```
SELECT revision FROM revisions WHERE lower(hex(updated)) = '"some hash value"';
```

So the issue is with `"some hash value"`. [Put a quote](https://github.com/magit/emacsql#operators) (`'`) before that and it should work. Or replace it with `$s1` and provide the value as an argument.

**Comment by martin5233** [comment]
Hi Jonas,

thanks for the quick reply. This did the trick, I hadn't read the documentation thoroughly enough :-)
